graphql_api/helpers/mutation.py

```diff
@@ -75,5 +75,22 @@ def authenticated_resolver(queried_owner, info, *args, **kwargs):
     return authenticated_resolver
 
 
+def require_shared_account_or_part_of_org(resolver):
+    def authenticated_resolver(queried_owner, info, *args, **kwargs):
+        current_user = info.context["request"].user
+        if (
+            current_user
+            and current_user.is_authenticated
+            and queried_owner
+            and queried_owner.account
+            and current_user in queried_owner.account.users.all()
```

Comment:
💯 💯 💯  
LMK if this is being slow, can optimize with 
- a prefetch for the Account and Users objs when you hit the db for the Owner obj
- doing `current_user in queried_owner.account.users.all()` in a different way

Reply:
`queried_owner` is the OwnerOrg that the `current_user` is checking permissions for right? Not the `current_user`'s OwnerUser object 🫠

Reply:
@nora-codecov Yes. Specifically this decorator should allow users to see fields on owners they share an account with. For example, I'm in codecov org and not sentry org. Since we share an Account, this decorator will let me see certain things on Sentry that an unrelated user could not see.